R_scripts/licordat.R

```diff
@@ -78,38 +78,48 @@ read_dir <- function(path) {
 collarDat <- read_csv("../design/cores_collars.csv")
+# `plotDat` holds information based on the plot code: longitude, latitude,
+# area, and salinity/elevation levels
 plots <- read_csv("../design/plots.csv")
 
+cat("Joining datasets and calculating...\n")
+
+# Merge these three datasets together based on collar number and plot name
 licorDat <- left_join(rawDat, collarDat, by = "Collar") %>% 
   rename(Origin_Plot = Plot) %>%
-  select(-Site)
-licorDat <- left_join(licorDat, plots, by = c("Origin_Plot" = "Plot")) %>%
+  select(-Site) %>% 
+  left_join(plots, by = c("Origin_Plot" = "Plot")) %>%
   rename(Origin_Salinity = Salinity, Origin_Elevation = Elevation) %>%
   select(-Site)
 
 # For any transplant core X, we know (in "Core_placement") the hole in which it ended up (or
 # rather, the core number of the hole). We actually need to know the plot. So create a lookup
 # table for this...
-lookup_table <- collarDat %>% 
-  select(Collar, Destination_Plot = Plot)
+lookup_table <- select(collarDat, Collar, Destination_Plot = Plot)
 
 # ...and then merge back into main data frame. Now "Lookup_Plot" holds the plot info for
 # where each core ENDED UP, not where it STARTED
 licorDat <- left_join(licorDat, lookup_table, by = c("Core_placement" = "Collar")) %>% 
   # Remove duplicate variables
-  select(-Longitude, -Latitude, -Plot_area_m2)
-licorDat <- left_join(licorDat, plots, by = c("Destination_Plot" = "Plot")) %>%
+  select(-Longitude, -Latitude, -Plot_area_m2) %>% 
+  left_join(plots, by = c("Destination_Plot" = "Plot")) %>%
   rename(Dest_Salinity = Salinity, Dest_Elevation = Elevation)
 
-# Reorder labels
-licorDat$Origin_Salinity <- factor(licorDat$Origin_Salinity, levels = c("High", "Medium", "Low"))
-licorDat$Origin_Elevation <- factor(licorDat$Origin_Elevation, levels = c("Low", "Medium", "High"))
-licorDat$Dest_Salinity <- factor(licorDat$Dest_Salinity, levels = c("High", "Medium", "Low"))
-licorDat$Dest_Elevation <- factor(licorDat$Dest_Elevation, levels = c("Low", "Medium", "High"))
-
-licorDat$Date <- paste(month(licorDat$Timestamp), "/", day(licorDat$Timestamp))
-licorDat$Group <- paste(licorDat$Origin_Plot, "->", licorDat$Destination_Plot)
-licorDat$Group[licorDat$Experiment == "Control"] <- "Control"
+# Reorder labels by making them into factors
+HML <- c("High", "Medium", "Low")
+licorDat %>% 
+  mutate(Origin_Salinity <- factor(Origin_Salinity, levels = HML),
+         Origin_Elevation <- factor(Origin_Elevation, levels = HML),
+         Dest_Salinity <- factor(Dest_Salinity, levels = HML),
+         Dest_Elevation <- factor(Dest_Elevation, levels = HML),
+         Date = paste(month(Timestamp), "/", day(Timestamp)),
+         Group = paste(Origin_Plot, "->", Destination_Plot),
+         Group = if_else(Experiment == "Control", "Control", Group)) ->
+  licorDat
 
-save(licorDat, file = "../outputs/licordat.rda")
\ No newline at end of file
+cat("Saving data...\n")
+write_csv(licorDat, "../outputs/licor_data.csv")
```

Comment:
@stephpenn1 - in general we want to save using human-readable formats like CSV, not RDA

Reply:
This was saved as an RDA to preserve the column formats. If you remember the issue I had ordering the plots using facet_grid.. we decided to save as RDA instead. I can save it as RDA and CSV?

Reply:
Hmm. Ah, OK, thanks. Let's save as both then; it's really useful to have a human-readable format.